lib/src/main/scala/spinal/lib/ethernet/Axi4StreamConditionalJoin.scala

```diff
@@ -0,0 +1,37 @@
+package spinal.lib.ethernet
+
+import spinal.core._
+import spinal.lib.Stream
+import spinal.lib.bus.amba4.axis.Axi4Stream.Axi4StreamBundle
+
+object Axi4StreamConditionalJoin {
+
+  /** Convert a tuple of streams into a stream of tuples
+    * source stream1 has higher priority
+    */
+  def apply[T1 <: Axi4StreamBundle, T2 <: Axi4StreamBundle](
```

Comment:
One way to deal wth the select is to stream1.haltWhen(!source1Select) , same for source2 :)

``` scala
    combined.valid := (sources(0).valid && sources(1).valid) |
      (source1Select && sources(0).valid)
    sources(0).ready := combined.fire
    sources(1).ready := source2Select & combined.fire
```

This is some kind of tricky arbitration. Maybe the utlity should be more neutral ? Or more documented about the edge case : ex !source1.valid && source2.valid && source2Select

Seems like this could be a regular Data instead of Axi4StreamBundle type paramater ?

Reply:
Maybe this utility should be more like : 

```scala
object SubStreamJoin{
  def apply[T1 <: Data, T2 <: Data](
      main: Stream[T1], sub: Stream[T2],
      withSub: Bool
  ): Stream[TupleBundle2[T1, T2]] = {
}
```

?

Where main stream can flow throw if !withSub
Else, sub has to be valid and will be merged

If the user want to hold the main stream, he can just use main.haltWhen(xxx) outside this utility

?

Reply:
You are right! 

Because the function I want to achieve here is to merge two streams in some condition, using one stream as primary and another one as a sub could be better.

I will improve it in the next commit.